tools/run-container

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2022 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import os
+import shlex
+import sys
+import tempfile
+
+from subprocess import run
+
+
+def logged(func):
+    def wrapper(args, **kwargs):
+        print('+', shlex.join(args))
+        return func(args, **kwargs)
+    return wrapper
+
+
+def git(*args):
+    run(['git', *args], check=True)
+
+
+def git_output(*args):
+    return run(['git', *args], check=True, capture_output=True, text=True).stdout.strip()
+
+
+def podman(*args, check=True):
+    if os.path.exists('/run/.toolboxenv'):
+        cmd = ['flatpak-spawn', '--host', 'podman', *args]
+    else:
+        cmd = ['podman', *args]
+
+    return run(cmd, check=check)
+
+
+class PodmanTemporaryDirectory(tempfile.TemporaryDirectory):
+    """TemporaryDirectory subclass capable of removing files owned by subuids"""
+    @classmethod
+    def _rmtree(cls, name, ignore_errors=False):
+        podman('unshare', 'rm', '-r', name)
```

Comment:
Unused -- supposedly this should be called on `__exit__()`?

Reply:
This is not unused.  It's overriding a protected method declared in `TemporaryDirectory()` and is the entire point of this subclass: we need "more power" than the normal `rm -rf` approach to this, so we use `podman`.